### DISCUSSION TITLE
Example from your documentation (Websocket)

### DISCUSSION THREAD

**Discussion by KH93b** [nanoexpress/nanoexpress#229] [General] [answered]
This example from your documentation doesn't work for me.
Have you tried running it?

import { exposeWebSocket } from 'nanoexpress/exposes/index.js';

app.ws('/ws', (req, res) => exposeWebSocket({
  console.log('Connecting...');

  res.on('connection', (ws) => {
    console.log('Connected');

    ws.on('message', (msg) => {
      console.log('Message received', msg);
      ws.send(msg);
    });
    ws.on('close', (code, message) => {
      console.log('Connection closed', { code, message });
    });
  });
  res.on('upgrade', () => {
    console.log('Connection upgrade');
  });
}));

**Comment by dalisoft** [comment, marked as answer]
Hi @KH93b
Thank you for testing and reporting.
Would be nice if you format your code, give more reproducible code and even better what happens? Echo example could be even better

**Comment by Ontopic** [comment]
Uhm, I'm pretty sure this is a clear mistake on the website still.

```
import { exposeWebSocket } from 'nanoexpress/exposes/index.js';

app.ws('/ws', (req, res) => exposeWebSocket({
  console.log('Connecting...');

  res.on('connection', (ws) => {
    console.log('Connected');

    ws.on('message', (msg) => {
      console.log('Message received', msg);
      ws.send(msg);
    });
    ws.on('close', (code, message) => {
      console.log('Connection closed', { code, message });
    });
  });
  res.on('upgrade', () => {
    console.log('Connection upgrade');
  });
}));
```

I assume it should either be;
```
app.ws('/ws', exposeWebSocket((req, res) => {
```
or
```
app.ws('/ws', (req, res) =>  exposeWebSocket((...args) => {
```

**Reply by dalisoft** [reply]
@Ontopic seems you using ultimate version. on main version no exposes yet
